File: Bubble-Sort/BubbleSort.kt
fun bubbleSort(array: Array<Int>): IntArray {
    for (i in 0..array.size - 2) {
        for (j in 0..array.size - i - 2) {
            if (array[j] > array[j + 1]) {
                val temp = array[j]
                array[j] = array[j + 1]
                array[j + 1] = temp

            }
        }
    }
    return array
}

fun main(args: Array<String>) {
    val unsorted = intArrayOf(1, 5, 4, 6, 8, 12, 9, 4, 5);
    val sorted = bubbleSort(unsorted)
    println("Before: " + unsorted)
    println("After: " + sorted)
}

File: Bubble-Sort/BubbleSort.c
#include <stdio.h>

void bubble_sort(int *array, int size) {
    for (int i = 0; i < size - 1; i++) {
        for (int j = 0; j < size - i - 1; j++) {
            if (array[j] > array[j + 1]) {
                int temp = array[j];
                array[j] = array[j + 1];
                array[j + 1] = temp;
            }
        }
    }
}

void print_array(int array[], int size) {
    for (int i = 0; i < size; i++) {
        printf("%d ", array[i]);
    }
    printf("\n");
}

int main() {
    int array[] = {5, 9, 8, 7, 6, 5, 4, 9, 12, 7};
    printf("Before:\n");
    print_array(array, 10);
    bubble_sort(array, 10);
    printf("After:\n");
    print_array(array, 10);

}
